sql/sql_table.cc

```diff
@@ -3948,27 +3957,36 @@ mysql_prepare_create_table(THD *thd, HA_CREATE_INFO *create_info,
 	  key_part_length= MY_MIN(column->length,
                                blob_length_by_type(sql_field->sql_type)
                                * sql_field->charset->mbmaxlen);
+
 	  if (key_part_length > max_key_length ||
-	      key_part_length > file->max_key_part_length())
+             key_part_length > file->max_key_part_length())
 	  {
-	    key_part_length= MY_MIN(max_key_length, file->max_key_part_length());
-	    if (key->type == Key::MULTIPLE)
-	    {
-	      /* not a critical problem */
-	      push_warning_printf(thd, Sql_condition::WARN_LEVEL_WARN,
-                                  ER_TOO_LONG_KEY,
-                                  ER_THD(thd, ER_TOO_LONG_KEY),
-                                  key_part_length);
-              /* Align key length to multibyte char boundary */
-              key_part_length-= key_part_length % sql_field->charset->mbmaxlen;
-	    }
-	    else
-	    {
-	      my_error(ER_TOO_LONG_KEY, MYF(0), key_part_length);
-	      DBUG_RETURN(TRUE);
-	    }
-	  }
-	}
+      /*currently Only InnoDB and MyISAM supports long unique constraints*/
```

Comment:
Instead of this you could add a comment like Allow long unique constraint if storage engine supports them. Making it like that when some other storage engine does add support there is no need to change this comment.